src/mongoose_acc.erl

```diff
@@ -1,300 +1,263 @@
 %%%-------------------------------------------------------------------
-%%% @doc
-%%% This module encapsulates a data type which will is instantiated when stanza
-%%% enters the system and is passed all the way along processing chain.
-%%% Its interface is map-like, and you can put there whatever you want, bearing in mind two things:
-%%% 1. It is read-only, you can't change value once you wrote it (accumulator is to accumulate)
-%%% 2. Whatever you put will be removed before the acc is sent to another c2s process
+%%% File    : mongoose_acc.erl
+%%% Author  : Piotr Nosek <piotr.nosek@erlang-solutions.com>
+%%% Author  : Bartlomiej Gorny <bartlomiej.gorny@erlang-solutions.com>
+%%% Purpose : Mongoose accumulator implementation
+%%% Created : 11 Sep 2018 by Piotr Nosek <piotr.nosek@erlang-solutions.com>
 %%%
-%%% There are three caveats to the above:
-%%% 1. Although you can not put to an existing key, you can append as many times as you like
-%%% 2. A special key 'result' is writeable and is meant to be used to get return value from hook calls
-%%% 3. If you want to pass something to another c2s process you can use add_prop/3 - values
-%%%    put there are not stripped.
-%%% @end
+%%% NS:Key conventions:
+%%% * hook:result should be used to return hook processing result
+%%% * iq:* contains useful IQ metadata but must be provided by mongoose_iq.erl
 %%%-------------------------------------------------------------------
 -module(mongoose_acc).
 -author("bartek").
```

Comment:
Like the famous big old oak "bartek" tree in Poland? :D

Reply:
I can add `-author("sokolica").` as well. :D